int32  mode
---
bool retval